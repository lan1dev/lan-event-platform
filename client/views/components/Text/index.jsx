import styled from 'styled-components';
import fonts from '../../constants/fonts';
import colors from '../../constants/colors';

/* eslint-disable react/destructuring-assignment */
const Text = styled.p`
  font-family: ${fonts.primary};
  font-size: ${props => (props.small ? '12px' : '14px')};
  display: ${props => (props.inline ? 'inline' : 'block')};
  color: ${props => (props.error ? colors.error : 'rgba(0, 0, 0, 0.65)')};
`;
/* eslint-enable react/destructuring-assignment */

export default Text;
