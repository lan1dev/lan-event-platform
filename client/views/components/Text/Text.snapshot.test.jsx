import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import Text from './index';

describe('Text', () => {
  let props;
  let wrapper;

  const Wrapper = () => {
    if (!wrapper) {
      wrapper = shallow(<Text {...props}>{props.children}</Text>);
    }
    return wrapper;
  };

  // Reset Component and Props before each test.
  beforeEach(() => {
    props = {};
    wrapper = undefined;
  });

  describe('without props', () => {
    it('Should match snapshot', () => {
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
  });

  describe('with small props', () => {
    it('Should match snapshot', () => {
      props = {
        small: true
      };
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
  });

  describe('with inline props', () => {
    it('Should match snapshot', () => {
      props = {
        inline: true
      };
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
  });

  describe('with error props', () => {
    it('Should match snapshot', () => {
      props = {
        error: true
      };
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
  });
});
