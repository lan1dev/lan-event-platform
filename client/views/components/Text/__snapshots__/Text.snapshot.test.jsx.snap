// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text with error props Should match snapshot 1`] = `
.c0 {
  font-family: Poppins;
  font-size: 14px;
  display: block;
  color: red;
}

<p
  className="c0"
/>
`;

exports[`Text with inline props Should match snapshot 1`] = `
.c0 {
  font-family: Poppins;
  font-size: 14px;
  display: inline;
  color: rgba(0,0,0,0.65);
}

<p
  className="c0"
/>
`;

exports[`Text with small props Should match snapshot 1`] = `
.c0 {
  font-family: Poppins;
  font-size: 12px;
  display: block;
  color: rgba(0,0,0,0.65);
}

<p
  className="c0"
/>
`;

exports[`Text without props Should match snapshot 1`] = `
.c0 {
  font-family: Poppins;
  font-size: 14px;
  display: block;
  color: rgba(0,0,0,0.65);
}

<p
  className="c0"
/>
`;
